### ISSUE TITLE
[MNG-5823] mvnDebug doesn't work with M2_HOME with spaces - missing quotes

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#7913]
**[Tobias Oberlies](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=tobias.oberlies)** opened **[MNG-5823](https://issues.apache.org/jira/browse/MNG-5823?redirect=false)** and commented

When calling `mvnDebug` in a Cygwin shell, I get the following error

```
Preparing to Execute Maven in Debug Mode
dirname: extra operand `Files/Maven/apache-maven-3.3.1/bin/mvnDebug'
Try `dirname --help' for more information.
env: /mvn: No such file or directory
```

The root cause are missing quotes in and around the dirname call. The script currently calls

```
env MAVEN_OPTS="$MAVEN_OPTS $MAVEN_DEBUG_OPTS" $(dirname $0)/mvn "$@"
```

but the call should be

```
env MAVEN_OPTS="$MAVEN_OPTS $MAVEN_DEBUG_OPTS" "$(dirname "$0")/mvn" "$@"
```

---

**Affects:** 3.3.1

**Issue Links:**
- [MNG-1244](https://issues.apache.org/jira/browse/MNG-1244) bin/m2 breaks with spaces in path

**Remote Links:**
- [Page
](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=67636510)

**Backported to:** [3.5.0-alpha-1](https://github.com/apache/maven/milestone/92?closed=1)

0 votes, 5 watchers

**Comment by jira-importer** [comment]
**[Stephen Connolly](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=stephenc)** commented

Maven 3.4.0 has been dropped. See [this thread](https://mail-archives.apache.org/mod_mbox/maven-dev/201701.mbox/%3CCA%2BnPnMx-e7kGYy3Hp87v8hLGdhp1q%3DtKLx_6QuZ4kGUqHEBGcw%40mail.gmail.com%3E) for more details.

This issue will need to be re-scheduled for a Maven release in the (hopefully near) future.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [46c1581474ed3b8110dbc7275172b92056b6e5dd](https://git-wip-us.apache.org/repos/asf?p=maven.git;a=commit;h=46c1581474ed3b8110dbc7275172b92056b6e5dd).
